src/ios/CDVContacts.m

```diff
@@ -597,7 +597,7 @@ - (void)createAddressBook:(CDVAddressBookWorkerBlock)workerBlock
     // NSLog(@"addressBook access: %lu", status);
     ABAddressBookRequestAccessWithCompletion(addressBook, ^(bool granted, CFErrorRef error) {
             // callback can occur in background, address book must be accessed on thread it was created on
-            dispatch_sync(dispatch_get_main_queue(), ^{
+            dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Did you check in the upstream repo if there was any rationale behind the implementation? It might be worth to check it and make sure that change doesn't introduce any unexpected behaviour.

Reply:
The last chage on this file on the upstream is from 2016 and this problem occurs on iOS 14. I think upstream is deprecated.

Reply:
We executed the tests today on iOS 13 and iOS 14 and both passed.

Reply:
Reading the comment on line 599, are we sure that line 596, where the address book is created, is executed on the main thread?

Reply:
I understand your point, but the create was not being forced into the main. Should I force it? in fact I'm not sure that create occurs on the main thread.

Reply:
Actually I think what @Chuckytuh should be done. So forcing the:

addressBook = ABAddressBookCreateWithOptions(NULL, &error);
ABAddressBookRequestAccessWithCompletion

To run on the main thread. This would disable any chance in the future to start this function in a thread other then the main.

Reply:
@vmfo22 by the comment I think this was added and tested when this function goes to background thread. But not retested on the main thread. If this block is called on the main thread will always deadlock.

Reply:
Changed to force ABAddressBookCreateWithOptions run on main thread.